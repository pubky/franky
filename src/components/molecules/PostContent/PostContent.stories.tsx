import type { Meta, StoryObj } from '@storybook/nextjs-vite';
import { PostContent } from './PostContent';

const meta: Meta<typeof PostContent> = {
  title: 'Molecules/Post/PostContent',
  component: PostContent,
  parameters: {
    layout: 'centered',
    backgrounds: {
      default: 'dark',
      values: [
        { name: 'dark', value: '#05050A' },
        { name: 'light', value: '#F2F2F7' },
      ],
    },
  },
  args: {
    text: 'I have said it before: If you don’t believe me or don’t get it, I don’t have time to try to convince you.',
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = { render: (args) => <PostContent {...args} /> };
