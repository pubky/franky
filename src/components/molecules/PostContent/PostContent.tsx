import * as React from 'react';
import * as Libs from '@/libs';

export interface PostContentProps {
  text: string;
  className?: string;
}

export function PostContent({ text, className }: PostContentProps) {
  return (
    <div className={Libs.cn('flex flex-col gap-3 rounded-2xl', className)}>
      <p className="text-base leading-6 font-medium text-secondary-foreground">{text}</p>
    </div>
  );
}
