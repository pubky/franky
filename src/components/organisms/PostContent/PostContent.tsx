'use client';

import { useLiveQuery } from 'dexie-react-hooks';
import * as Core from '@/core';
import * as Libs from '@/libs';

export interface PostContentOrganismProps {
  postId: string;
  className?: string;
}

export function PostContent({ postId, className }: PostContentOrganismProps) {
  // Fetch post details for content
  const postDetails = useLiveQuery(async () => {
    return await Core.PostDetailsModel.findById(postId);
  }, [postId]);

  if (!postDetails) {
    // TODO: Add skeleton loading component for PostContent
    return <div className="text-muted-foreground">Loading content...</div>;
  }

  return (
    <div className={Libs.cn('flex flex-col gap-3 rounded-2xl', className)}>
      <p className="text-base leading-6 font-medium text-secondary-foreground break-all">{postDetails.content}</p>
    </div>
  );
}
