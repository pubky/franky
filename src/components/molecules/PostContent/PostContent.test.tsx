import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { PostContent } from './PostContent';

describe('PostContent', () => {
  it('renders text', () => {
    render(<PostContent text="Hello world" />);
    expect(screen.getByText('Hello world')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <PostContent text="I have said it before: If you don’t believe me or don’t get it, I don’t have time to try to convince you." />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
