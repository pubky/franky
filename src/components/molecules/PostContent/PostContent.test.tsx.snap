// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostContent > matches snapshot 1`] = `
<div
  class="flex flex-col gap-3 rounded-2xl"
>
  <p
    class="text-base leading-6 font-medium text-secondary-foreground"
  >
    I have said it before: If you don’t believe me or don’t get it, I don’t have time to try to convince you.
  </p>
</div>
`;
